examples/server/main.go

```diff
@@ -0,0 +1,172 @@
+						id := p.Args["id"].(int)
+						return documents[id], nil
+					},
+				},
+			},
+		},
+	)
+
+	var mutationType = graphql.NewObject(
+		graphql.ObjectConfig{
+			Name: "Mutation",
+			Fields: graphql.Fields{
+				"updateDocument": &graphql.Field{
+					Type: documentType,
+					Args: graphql.FieldConfigArgument{
+						"id": &graphql.ArgumentConfig{
+							Type: graphql.NewNonNull(graphql.Int),
+						},
+						"title": &graphql.ArgumentConfig{
+							Type: graphql.NewNonNull(graphql.String),
+						},
+						"content": &graphql.ArgumentConfig{
+							Type: graphql.NewNonNull(graphql.String),
+						},
+					},
+					Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+						id := p.Args["id"].(int)
+						documents[id].Title = p.Args["title"].(string)
+						documents[id].Content = p.Args["content"].(string)
+
+						for _, subs := range subscriptionManager.Subscriptions() {
+							for _, sub := range subs {
+								// JSON interface is float64
+								var subID int = int(sub.Variables["id"].(float64))
+
+								if id == subID {
+									params := graphql.Params{
+										Schema:         schema,
+										RequestString:  sub.Query,
+										VariableValues: sub.Variables,
+										OperationName:  sub.OperationName,
+									}
+									result := graphql.Do(params)
+
+									data := graphqlws.DataMessagePayload{
+										Data: result.Data,
+										Errors: graphqlws.ErrorsFromGraphQLErrors(
+											result.Errors,
+										),
+									}
+
+									sub.SendData(&data)
+								}
+							}
+						}
+
+						return documents[id], nil
+					},
+				},
+			},
+		},
+	)
+
+	var subscriptionType = graphql.NewObject(
+		graphql.ObjectConfig{
+			Name: "Subscription",
+			Fields: graphql.Fields{
+				"documentUpdates": &graphql.Field{
+					Type: documentType,
+					Args: graphql.FieldConfigArgument{
+						"id": &graphql.ArgumentConfig{
+							Type: graphql.Int,
+						},
+					},
+					Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+						id := p.Args["id"].(int)
+						return documents[id], nil
+					},
+				},
+			},
+		},
+	)
+
+	schemaConfig := graphql.SchemaConfig{
+		Query:        queryType,
+		Mutation:     mutationType,
+		Subscription: subscriptionType,
+	}
+
+	var err error
+	schema, err = graphql.NewSchema(schemaConfig)
+	if err != nil {
+		log.WithField("err", err).Panic("GraphQL schema is invalid")
+	}
+
+	subscriptionManager = graphqlws.NewSubscriptionManager(&schema)
+	websocketHandler := graphqlws.NewHandler(graphqlws.HandlerConfig{
+		SubscriptionManager: subscriptionManager,
+		Authenticate: func(token string) (interface{}, error) {
+			return "Default user", nil
+		},
+	})
+
+	graphqlHandler := handler.New(&handler.Config{
+		Schema:                &schema,
+		Pretty:                true,
+		GraphiQL:              true,
+		Endpoint:              "http://localhost:8085",
+		SubscriptionsEndpoint: "ws://localhost:8085/subscriptions",
```

Comment:
In `github.com/graphql-go/handler`, where `handler.Config` is defined, I don't think `Endpoint` or `SubscriptionsEndpoint` are properties.

Reply:
See https://github.com/graphql-go/handler/blob/master/handler.go#L162

Reply:
Any idea how to tell graphql about the subscriptions endpoint?

Reply:
You will also need this PR https://github.com/graphql-go/handler/pull/33/files